bezier-rs/lib/src/lib.rs

```diff
@@ -602,61 +602,115 @@ impl Bezier {
 		self.apply_transformation(&|point| point + translation)
 	}
 
-	// TODO: Use an `impl Iterator` return type instead of a `Vec`
-	// TODO: Change this to `intersect(&self, other: &Bezier)` to also work on quadratic and cubic segments
-	// TODO: (or keep this and add two more functions that perform the logic, and make the `intersect` function call the correct one)
-	/// Returns a list of points where the provided line segment intersects with the Bezier curve. If the provided segment is colinear with the bezier, zero intersection points will be returned.
-	/// - `line` - A line segment expected to be received in the format of `[start_point, end_point]`.
-	pub fn intersect_line_segment(&self, line: [DVec2; 2]) -> Vec<DVec2> {
-		// Rotate the bezier and the line by the angle that the line makes with the x axis
-		let slope = line[1] - line[0];
-		let angle = slope.angle_between(DVec2::new(1., 0.));
-		let rotation_matrix = DMat2::from_angle(angle);
-		let rotated_bezier = self.apply_transformation(&|point| rotation_matrix.mul_vec2(point));
-		let rotated_line = [rotation_matrix.mul_vec2(line[0]), rotation_matrix.mul_vec2(line[1])];
-
-		// Translate the bezier such that the line becomes aligned on top of the x-axis
-		let vertical_distance = rotated_line[0].y;
-		let translated_bezier = rotated_bezier.translate(DVec2::new(0., -vertical_distance));
-
-		// Compute the roots of the resulting bezier curve
-		let list_intersection_t = match translated_bezier.handles {
-			BezierHandles::Linear => {
-				// If the transformed linear bezier is on the x-axis, `a` and `b` will both be zero and `solve_linear` will return no roots
-				let a = translated_bezier.end.y - translated_bezier.start.y;
-				let b = translated_bezier.start.y;
-				utils::solve_linear(a, b)
+	/// Implementation of the algorithm to find curve intersections by iterating on bounding boxes.
+	/// `curve1_t_interval` is used to identify the t values of the original `curve1` that the current iteration is representing.
+	/// Note that the `t` interval for only the first curve is needed since we want to return `t` with respect to it.
+	fn intersections_between_subcurves(curve1: &Bezier, original_curve1_t_interval: [f64; 2], curve2: &Bezier, error: f64) -> Vec<f64> {
```

Comment:
Shouldn't this take `&self` instead of two `&Bezier`s? Then you can call `curve2` "`other`".
Please also document what `original_curve1_t_interval` (or what it ends up being called, perhaps "self" or "own" instead of "curve1" in the name) means/does.